ante/fee.go

```diff
@@ -54,16 +86,52 @@ func (mfd BypassMinFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 				fee := gp.Amount.Mul(glDec)
 				requiredFees[i] = sdk.NewCoin(gp.Denom, fee.Ceil().RoundInt())
 			}
-
-			if !feeCoins.IsAnyGTE(requiredFees) {
+			// 1stake is not a subset of 0stake or 0photon or 1photon, skip the min gas price check
+			// can still the DenomsSubsetOf() from sdk
+			if !feeCoins.DenomsSubsetOf(requiredFees.Sort()) {
+				return next(ctx, tx, simulate)
+			}
+			// requiredFees here is ensured not all zero, when check min_gas_price, fee might be zero. if min_gas_price=0stake,1photon, and feecoins is 0stake, it should not return err. so use IsAnyGTE() rather than IsAnyGTE() from sdk.
+			if !IsAnyGTE(feeCoins, requiredFees) {
 				return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins, requiredFees)
 			}
 		}
 	}
 
+	// when the tx is bypass msg type, still need to check the denom is not random denom
+	// this is to prevent the situation that a bypass msg carries random fee denoms
+	if ctx.IsCheckTx() && !simulate && (mfd.bypassMinFeeMsgs(msgs) && gas <= uint64(len(msgs))*maxBypassMinFeeMsgGasUsage) && mfd.GlobalMinFee.Has(ctx, types.ParamStoreKeyMinGasPrices) {
+		requiredFees := mfd.getGlobalFee(ctx, feeTx)
+		if !DenomsSubsetOf(feeCoins, requiredFees) {
+			return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "fees denom is wrong; got: %s required: %s", feeCoins, requiredFees)
+		}
+	}
+
 	return next(ctx, tx, simulate)
 }
 
+//coins must be sorted
+func (mfd BypassMinFeeDecorator) getGlobalFee(ctx sdk.Context, feeTx sdk.FeeTx) sdk.Coins {
+	var globalMinGasPrices sdk.DecCoins
+	mfd.GlobalMinFee.Get(ctx, types.ParamStoreKeyMinGasPrices, &globalMinGasPrices)
+
+	// global fee is empty set, set global fee to 0uatom
+	if len(globalMinGasPrices) == 0 {
+		globalMinGasPrices = defaultZeroGlobalFee
```

Comment:
don't forget to replace with `DefaultZeroGlobalFee` if that function is used instead.